Test/implementationDivision/focus/focus.bpl.expect

```diff
@@ -1,3 +1,118 @@
+implementation Ex() returns (y: int)
+{
+
+  anon0:
+    assert 5 + 0 == 5;
+    assert 5 * 5 <= 25;
+    assume x#AT#0 + y#AT#0 == 5;
+    assume x#AT#0 * x#AT#0 <= 25;
+    assume {:partition} 0 >= x#AT#0;
+    assume y#AT#2 == y#AT#0;
+    assert y#AT#2 >= 0;
+    return;
+}
+
+
+implementation Ex/split@15() returns (y: int)
```

Comment:
It feels confusing that you end up with three different patterns for naming splits here: 

1. The first unqualified `Ex` split above (and what does that cover then?)
2. This "number 15" split
3. The path-based splits below

Being clear on this matters since the user has to understand their source in order to fix verification failures.

I would at least suggest being slightly clearer on the path-based labels, perhaps something like `Ex[path: 14 -> 19 -> 25]`. You could produce that in Dafny error messages or at the Boogie level.

Reply:
> The first unqualified Ex split above (and what does that cover then?)

Impossible to say what it covers. It is the "remainder" after processing all the focus/isolate/split_here

> This "number 15" split

It's actually the line number from which the assertion originates. I could add the column to possibly disambiguate

> Being clear on this matters since the user has to understand their source in order to fix verification failures.

This notation is only used at the Boogie level, where I optimize for power users. I use more verbose notation in Dafny.

Reply:
I've updated the Boogie naming so it's simply more correct and also slightly more verbose. It will say `implementation Ex/focus[16]/split@15()` to mean, the implementation Ex, using the `focus` at line 16, and then the `split_here` at line 15.

One thing I struggle with is the overloading of the term split. Split is used to describe both any type of VC partitioning: `focus`/`split_here`/`isolate`/`isolate "paths"`, but also occurs in the name "split_here". Should I rename `{:split_here}` to `{:cut}` ?

Reply:
Naming is much better, thanks!

Heavily agreed on the overloading of "split" - it fits better as an operation name rather than a name for the things it produces. I think `{:split_here}` is reasonable since it's referring to the operation, it's labelling the VCs as "splits" that's a bigger problem.

Reply:
I initially wanted to rename the produced things to `Partition`, but somehow I kept referring to things as splits. I feel it would be easiest to rename `{:split_here}` to `{:cut}`, but I'm willing to rename on the other side.